src/components/testCaseLanding/qdm/TestCaseCoverage/CoverageTabs/CoverageTabList.tsx

```diff
@@ -1,110 +1,115 @@
-              (statementResult) =>
-                Object.keys(statementResult)?.filter((definitionName) =>
-                  getDefintionCategoryFilteringCondition(
-                    definitionType,
-                    statementResult,
-                    definitionName
-                  )
-                )
-            )
-          )
-        )
-      ),
-    ];
+  }, [testCaseGroups, populationCriteria.id, getPopulationAbbreviation]);
 
-    return definitionNames.reduce((acc, definitionName) => {
-      if (mappedCql.definitions[definitionName]) {
-        acc[definitionName] = mappedCql.definitions[definitionName];
+  const getStatementResultsInCategory = (statementResults, definition) => {
+    if (statementResults) {
+      if (definition === "Used") {
+        return statementResults.filter((s) => s.relevance !== "NA");
       }
-      return acc;
-    }, {});
-  };
-
-  const getDefinitionResults = (definitionType) => {
-    if (
-      mappedCql &&
-      mappedCql.functions &&
-      mappedCql.definitions &&
-      calculationOutput
-    ) {
-      if (definitionType === "Functions") {
-        return mappedCql.functions;
-      }
-      if (definitionType === "Unused") {
-        return filterBasedOnDefinitionCategories(definitionType);
+      if (definition === "Unused") {
+        return statementResults.filter((s) => s.relevance === "NA");
       }
-      if (definitionType === "Used") {
-        return filterBasedOnDefinitionCategories(definitionType);
+      if (definition === "Functions") {
+        return statementResults.filter((s) => s.type === "FunctionDef");
       }
     }
-    return null;
+    return [];
   };
 
   return (
     <div data-testid="coverage-tab-list">
-      {mappedCql &&
-        populationCriteria.length &&
-        populationCriteria.map((pop, i) => {
+      {relevantPops &&
+        relevantPops.map((pop, i) => {
+          const selectedGroup = measureGroups?.find(
+            (group) => group.id === populationCriteria.id
+          );
+          const selectedPopulation = selectedGroup?.populations?.find(
+            (s) => s.id === pop.id
+          );
+          // get all clauses that match with definition (e.i Initial Population)
+          const coverage =
+            groupCoverageResult &&
+            groupCoverageResult[populationCriteria.id]?.find(
+              (coverageResult) =>
+                coverageResult.name === selectedPopulation?.definition
+            );
+
           return (
             <CoverageTab
               key={i}
-              population={getPopulationAbbreviation(
-                populationCriteria,
-                pop.name,
-                i
-              )}
-              definitionText={mappedCql?.populationDefinitions[pop?.name]}
+              cqlDefinitionCallstack={cqlDefinitionCallstack}
+              definition={startCase(pop.name.split(/(?=[A-Z])/).join(" "))}
+              groupCoverageResult={groupCoverageResult}
```

Comment:
you need to pass results for only selected group because coverage is by group 
```suggestion
              groupCoverageResult={groupCoverageResult[populationCriteria.id]}
```